devices/src/virtio/block.rs

```diff
@@ -158,16 +146,46 @@ struct Request {
     status_addr: GuestAddress,
 }
 
+/// The request header represents the mandatory fields of each block device request.
+///
+/// A request header contains the following fields:
+///   * request_type: an u32 value mapping to a read, write or flush operation.
+///   * reserved: 32 bits are reserved for future extensions of the Virtio Spec.
+///   * sector: an u64 value representing the offset where a read/write is to occur.
+///
+/// The header simplifies reading the request from memory as all request follow
+/// the same memory layout.
+#[derive(Copy, Clone, Default, Debug)]
+#[repr(C)]
+struct RequestHeader {
```

Comment:
I suggest a limiting the target endianess for this struct, so if we ever decide to support big endian, this wouldn't compile.

Reply:
I added the configure on the `read_from` function.